autoreduce_qp/queue_processor/reduction/process_manager.py

```diff
@@ -27,33 +25,89 @@ def __init__(self, message: Message, run_name: str) -> None:
+            serialized_vars = self.message.serialize()
+            serialized_vars_truncated = self.message.serialize(limit_reduction_script=True)
+            args = ["python3", "runner.py", serialized_vars, self.run_name]
+            logger.info("Calling: %s %s %s %s ", "python3", "runner.py", serialized_vars_truncated, self.run_name)
+
+            # Return a client configured from environment variables
+            # The environment variables used are the same as those used by the Docker command-line client
+            # https://docs.docker.com/engine/reference/commandline/cli/#environment-variables
+            client = docker.from_env()
+
+            # Create a container and run it. Equivalent to docker run.
+            # To build runner-mantid image, in same directory as Dockerfile run:
+            # docker build -t runner-mantid .
 
-                # Copy and update the subprocess environment to inherit the
-                # parent one, and append the PYTHONPATH of the queue_processor module
-                environment = os.environ.copy()
+            # Pull image, containers.create doesn't pull everytime
+            client.images.pull('ghcr.io/autoreduction/runner-mantid:6.2.0')
+            mount = f'{os.path.expanduser("~")}/.autoreduce/dev/test-archive'
+            if not os.path.exists(mount):
+                Path(mount).mkdir(parents=True, exist_ok=True)
+            os.chmod(mount, 0o777)
 
-                pythonpath = environment.get("PYTHONPATH", "")
-                pythonpath += f':{RUNNER_PATH.split("autoreduce_qp")[0]}'
-                pythonpath = environment["PYTHONPATH"] = pythonpath
+            Path(f'{os.path.expanduser("~")}/.autoreduce/dev/reduced-data').mkdir(parents=True, exist_ok=True)
+            os.chmod(f'{os.path.expanduser("~")}/.autoreduce/dev/reduced-data', 0o777)
 
-                # Run process until finished and check the exit code for success
-                subprocess.run(args, check=True, env=environment)
+            container = client.containers.create(
+                image="ghcr.io/autoreduction/runner-mantid:6.2.0",
+                command="/bin/sh",
+                volumes={
+                    f'{os.path.expanduser("~")}/.autoreduce/': {
+                        'bind': f'{os.path.expanduser("~")}/.autoreduce/',
+                        'mode': 'rw'
+                    },
+                    mount: {
+                        'bind': '/isis/',
+                        'mode': 'rw'
+                    },
+                    f'{os.path.expanduser("~")}/.autoreduce/dev/reduced-data': {
+                        'bind': '/instrument/',
+                        'mode': 'rw'
+                    },
+                    temp_dir.name: {
+                        'bind': '/output/',
+                        'mode': 'rw'
+                    },
+                },
+                tty=True,
+                environment=["AUTOREDUCTION_PRODUCTION=1"],
```

Comment:
Does this need to be added? I don't think the runner containers should care whether it's production or not, as the only thing that changes is the mount src paths